src/zarr/codecs/_v2.py

```diff
@@ -68,6 +68,9 @@ async def _encode_single(
     ) -> Buffer | None:
         chunk = chunk_array.as_ndarray_like()
 
+        # ensure contiguous and correct order
+        chunk = chunk.astype(chunk_spec.dtype, order=chunk_spec.order)
```

Comment:
```suggestion
        chunk = chunk.astype(chunk_spec.dtype, order=chunk_spec.order, copy=False)
```

This should only copy if needed